background.js

```diff
@@ -1,128 +1,374 @@
+				console.log('copyToClip');
+				rcxMain.copyToClip(sender.tab, request.entry);
+				break;
+			default:
+				console.log(request);
+		}
+	})
+
+var optionsList = [
+	"copySeparator", // Is this field separator?
+	"disablekeys",
+	"highlight",
+	"kanjicomponents",
+	"kanjiInfo",
+	"lineEnding",
+	"maxClipCopyEntries",
+	"minihelp",
+	"onlyreading",
+	"popupcolor",
+	"popupDelay",
+	"popupLocation",
+	"textboxhl",
+	"showOnKey"];
+
+var kanjiInfoLabelList = rcxDict.prototype.kanjiInfoLabelList;
+this.kanjiInfoObject = {}; //Object used to save kanjiinfo to cloud.
+rcxMain.config = {};
+
+/** Get option data from cloud. */
+chrome.storage.sync.get(optionsList,
+	function (items) {
+		fillRcxValsFromLocalStoreCloudAndDefaultValues(items);
+		saveOptionsToCloudStorage();
+	})
+
+
+/**
+ * Initializes the localStorage for the given key.
+ * If the given key is already initialized, nothing happens.
+ *
+ * @author Teo (GD API Guru)
+ * @param key The key for which to initialize
+ * @param initialValue Initial value of localStorage on the given key
+ * @return true if a value is assigned or false if nothing happens
+ */
+function initStorage(key, initialValue) {
+	var currentValue = localStorage[key];
+	if (!currentValue) {
+		localStorage[key] = initialValue;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * Populates config with option values from 3 places
+ * 1. Cloud Storage, 2. Local Storage, and 3. Default
+ * If any values are in local storage they will overwrite cloud storage values.
+ * Values should no longer be saved to local storage.
+ *
+ * @author Deshaun (From Dev Color)
+ * @param items option values retrieved from cloud storage
+ * @return None
+ */
+function fillRcxValsFromLocalStoreCloudAndDefaultValues(items) {
+	/*
+		For Each Option Value below,
+		Condition 1: Cloud Storage Value
+		Condition 2: Local Storage Value
+		Condition 3: Default Value
+	 */
+
+	if (initStorage("v0.8.92", true)) {
+		/** Useful old versioning logic */
+		// 	if (initStorage("v0.8.92", true)) {
+		// 		// v0.7
+		// 		// initStorage("popupcolor", "blue");
+		// 		initStorage("highlight", true);
+		//
+		// 		// v0.8
+		// 		// No changes to options
+		//
+		// 		// V0.8.5
+		// 		initStorage("textboxhl", false);
+		//
+		//
+		// });
+
+		// v0.8.8
+		// Backward compatibility logic may drop soon.
+		if (localStorage['highlight'] == "yes")
+			localStorage['highlight'] = "true";
+		if (localStorage['highlight'] == "no")
+			localStorage['highlight'] = "false";
+		if (localStorage['textboxhl'] == "yes")
+			localStorage['textboxhl'] = "true";
+		if (localStorage['textboxhl'] == "no")
+			localStorage['textboxhl'] = "false";
+		if (localStorage['onlyreading'] == "yes")
+			localStorage['onlyreading'] = "true";
+		if (localStorage['onlyreading'] == "no")
+			localStorage['onlyreading'] = "false";
+	}
+
+	//Todo: Need to Organize these options in alphabetical order.
+	/** Set Copy Separator Option value */
+	if (items.copySeparator) {
```

Comment:
instead of 3 lines for every I think we can just modify initStorage to do these checks.

We also need to be careful about type differences since local storage is all strings while cloud sync can have real types.